inst/doctest.m

```diff
@@ -160,6 +160,19 @@
 %% @end group
 %% @end example
 %%
+%% You can also mark tests depending on runtime conditions, such as the
+%% platform, operating systems, or installed packages:
+%%
+%% @example
+%% @group
+%% >> "some shiny octave feature"         % doctest: +XFAIL_IF(DOCTEST_MATLAB)
+%% ans = some shiny octave feature
+%% @end group
+%% @end example
+%%
+%% Doctest provides the default flags DOCTEST_OCTAVE and DOCTEST_MATLAB,
+%% but you can access arbitrary variables or functions.
+%%
```

Comment:
Right now, you can only access variables and nullary functions (like `is_octave`) thanks to their syntactic equivalence. I'd be happy to take a patch to support more general expressions, but it significantly complicates the parsing unless there is some builtin parser that we could use (e.g., can't parse nested parentheses using regular expressions).